bazel/api_binding.bzl

```diff
@@ -0,0 +1,35 @@
+def _default_envoy_api_impl(ctx):
+    ctx.file("WORKSPACE", "")
+    ctx.file("BUILD.bazel", "")
+    api_dirs = [
+        "bazel",
+        "docs",
+        "envoy",
+        "examples",
+        "test",
+        "tools",
+    ]
+    for d in api_dirs:
+        ctx.symlink(ctx.path(ctx.attr.api).dirname.get_child(d), d)
+
+_default_envoy_api = repository_rule(
+    implementation = _default_envoy_api_impl,
+    attrs = {
+        "api": attr.label(default = "@envoy//api:BUILD"),
+    },
+)
+
+def envoy_api_binding():
+    # Treat the data plane API as an external repo, this simplifies exporting the API to
+    # https://github.com/envoyproxy/data-plane-api.
+    if "envoy_api" not in native.existing_rules().keys():
+        _default_envoy_api(name = "envoy_api")
+
+    native.bind(
```

Comment:
Do we need to still do this bind? Can we replace the bind uses inside the API tree with the canonical `@com_google_apis//...` labels? Same below.

Reply:
This would be fine as long as none of downstream envoy repositories refer to the bind. I don't have confidence that is the case, so I left it as-is for backwards compatibility reasons.

Reply:
I'm happy to say "go ahead" on this, we don't provide stability guarantees around WORKSPACE and you're moving towards the long-term stable canonical naming convention in Bazel. 

This is where recursive workspaces would be great.

Reply:
Unfortunately, grpc_json_transcoder depends on this binding, so I can't remove it without making changes to that repository. Can this be done later since we'll have to make changes to other repositories first?

Reply:
Ack, can you file an issue for cleanup here and tag the site with TODO?

Reply:
Filed https://github.com/envoyproxy/envoy/issues/7719 and referenced it here.